ERCS/erc-7554.md

```diff
@@ -0,0 +1,400 @@
+        uint256 tokenId,
+        uint256 benefitId,
+        benefitMetadataURI string
+    ) external;
+
+    /**
+     * @notice updates an attached benefit.
+     * Throws if `msg.sender` is not an approved address or owner of the contract.
+     * @param benefitId The unique identifier of the benefit to be updated.
+     * @param benefitMetadataURI the new url to updated benefit metadata json object
+     */
+    function updateBenefit(
+        uint256 benefitId,
+        benefitMetadataURI string
+    ) external;
+
+    /**
+     * @notice removes an attached benefit.
+     * Throws if `msg.sender` is not an approved address or owner of the contract.
+     * @param benefitId The unique identifier of the benefit to be removed.
+     */
+    function removeBenefit(
+        uint256 benefitId
+    ) external;
+
+    /**
+     * @notice Return true or false, if the provided wallet address is the assigner of the benefit.
+     * @param benefitId The unique identifier of the benefit.
+     * @return returns whether the the wallet address is the assigner of the benefit or not.
+     */
+    function isBenefitAssigner(
+        address wallet,
+        uint256 benefitId
+    ) external view returns(bool);
+
+    /**
+     * @notice Return metadata URI of the attached benefit.
+     * @param benefitId The unique identifier of the benefit.
+     * @return returns the Uniform Resource identifier URI for the `benefitId` metadata.
+     */
+    function benefitURI(
+        uint256 benefitId
+    ) external view returns(string memory);
+
+    /**
+     * @notice Return the list of benefitIds that are attached to a certain `tokenId`.
+     * @param tokenId The unique identifier of the NFT to get its list of assigned benefits.
+     * @return returns the list of benefits Ids.
+     */
+    function assignedBenefits(
+        uint256 tokenId
+    ) external view returns(uint256[] memory);
+
+    /**
+     * @notice Return the list of benefitIds that are attached to the collection not only a specific `tokenId`.
+     * @return returns the list of benefits Ids.
+     */
+    function assignedBenefits() external view returns(uint256[] memory);
+}
+```
+
+A contract implementing this standard MAY set a maximum number of benefits to be attached to any token in the collection, in that case the implementation should keep track of the total number of benefits attached to a single `tokenId` and this number MUST NOT be greater than the maximum number of benefits.
+
+The contract owner MAY set the `attachBenefit` as payable to be able to collect fees in advance and on-chain from interested parties in attaching benefits to the collection tokens.
+
+The `BenefitAttached` event MUST be emitted when `attachBenefit` is called.
+
+The `BenefitUpdated` event MUST be emitted when `updateBenefit` is called.
+
+The `BenefitRemoved` event MUST be emitted when `removeBenefit` is called.
+
+The `supportsInterface` method MUST return `true` when called with `TBD`.
```

Comment:
```suggestion
The `supportsInterface` method MUST return `true` when called with `TBD`. <!-- TODO -->
```

If you use HTML-style comments, the linter will warn you if you attempt to go to review without removing them.